src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs

```diff
@@ -33,9 +33,11 @@ public static Task<T> ReadFromJsonAsync<T>(this HttpContent content, JsonSeriali
 
         private static async Task<object?> ReadFromJsonAsyncCore(HttpContent content, Type type, Encoding? sourceEncoding, JsonSerializerOptions? options, CancellationToken cancellationToken)
         {
-            #pragma warning disable CA2016 // The overload that takes a CancellationToken is not available in netstandard2.0
+#if NETCOREAPP
+            Stream contentStream = await content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
+#else
             Stream contentStream = await content.ReadAsStreamAsync().ConfigureAwait(false);
-            #pragma warning restore
+#endif
```

Comment:
We generally prefer to use alternatives rather than sprinking ifdefs around the code.  For example, you can create an extension method that's only compiled into the netstandard build.
```C#
public static Task<Stream> ReadAsStreamAsync(this HttpContent content, CancellationToken cancellationToken) =>
    content.ReadAsStreamAsync(); // ignore cancellationToken due to lack of available API in netstandard2.0
```
Here's an example of that in this project:
https://github.com/dotnet/runtime/blob/1c09c34a706728fc87ec562dcf2c53ae801bb1c7/src/libraries/System.Net.Http.Json/src/System.Net.Http.Json.csproj#L21

Reply:
Thanks, done. There were other ifdef sections unrelated to this change, I moved them to the new split files.